tests/test_esquery.py

```diff
@@ -0,0 +1,417 @@
+                                'name3': 'value3'
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+        self.assertDictEqual(self.es._ElasticQuery__get_query_basic(self.date_field, self.start, self.end,
+                             self.filters).to_dict(), test_with_filters)
+
+    def test_query_aggregations(self):
+        """Test all the aggregation functions"""
+
+        # ------ Terms aggregation ------
+        test_terms_dict = {
+            "terms": {
+                "field": "AGG_FIELD",
+                "size": 100,
+                "order": {
+                    "_count": "desc"
+                }
+            }
+        }
+        agg_id, terms_agg = self.es._ElasticQuery__get_query_agg_terms(self.field, 22)
+        # check the output dict
+        self.assertDictEqual(test_terms_dict, terms_agg.to_dict())
+        # check the cutom agg id
+        self.assertEqual(agg_id, 22)
+
+        # ------ Max aggregation ------
+        test_max_dict = {
+            "max": {
+                "field": "AGG_FIELD"
+            }
+        }
+        _, max_agg = self.es._ElasticQuery__get_query_agg_max(self.field)
+        # check the output dict
+        self.assertDictEqual(test_max_dict, max_agg.to_dict())
+
+        # ------ Avg aggregation ------
+        test_avg_dict = {
+            "avg": {
+                "field": "AGG_FIELD"
+            }
+        }
+        _, avg_agg = self.es._ElasticQuery__get_query_agg_avg(self.field)
+        # check the output dict
+        self.assertDictEqual(test_avg_dict, avg_agg.to_dict())
+
+        # ------ Percentile aggregation ------
+        test_percentiles_dict = {
+            "percentiles": {
+                "field": "AGG_FIELD"
+            }
+        }
+        _, percentiles_agg = self.es._ElasticQuery__get_query_agg_percentiles(self.field)
+        # check the output dict
+        self.assertDictEqual(test_percentiles_dict, percentiles_agg.to_dict())
+
+        # ------ Cardinality aggregation ------
+        test_cardinality_dict = {
+            "cardinality": {
+                "field": "AGG_FIELD",
+                "precision_threshold": self.es.ES_PRECISION
+            }
+        }
+        _, cardinality_agg = self.es._ElasticQuery__get_query_agg_cardinality(self.field)
+        # check the output dict
+        self.assertDictEqual(test_cardinality_dict, cardinality_agg.to_dict())
+
+    def test_get_bounds(self):
+        """Test the get_bounds function"""
+
+        # with empty parameters
+        self.assertEquals(self.es._ElasticQuery__get_bounds(), {})
+
+        test_bounds_dict = {
+            "extended_bounds": {
+                "min": 1495497600000.0,
+                "max": 1527033600000.0
+            }
+        }
+        self.assertDictEqual(self.es._ElasticQuery__get_bounds(self.start, self.end), test_bounds_dict)
+
+    def test_get_query_agg_ts(self):
```

Comment:
I would split this test in two: tests that check exceptions and tests that do not